File: parallel.cpp
#include <chrono>
#include <cmath>
#include <future>
#include <iostream>
#include <vector>

double f(int n) {
    double s = 0;
    for (int i = 0; i < n; ++i) s += std::sqrt(double(i));
    return s;
}

int main() {
    using Clock = std::chrono::steady_clock;
    std::vector<int> inputs(8, 500'000'000);
    std::vector<double> a, b;

    auto start = Clock::now();
    for (int n : inputs) a.push_back(f(n));
    double t1 = std::chrono::duration<double>(Clock::now() - start).count();

    start = Clock::now();
    std::vector<std::future<double>> tasks;
    for (int n : inputs)
        tasks.push_back(std::async(std::launch::async, f, n));
    for (auto& task : tasks) b.push_back(task.get());
    double tp = std::chrono::duration<double>(Clock::now() - start).count();

    std::cout << "Sequential: " << t1 << " s\n";
    std::cout << "Parallel: " << tp << " s\n";
    std::cout << "Speedup: " << t1 / tp << " s\n";
    std::cout << "Same results: " << std::boolalpha << (a == b) << '\n';
}

File: parallel.swift
import Dispatch
import Synchronization

func f(_ n: Int) -> Double {
    var s = 0.0
    for i in 0..<n { s += Double(i).squareRoot() }
    return s
}

func demo() {
    let inputs = Array(repeating: 5_000_000, count: 8)
    let results = Mutex(Array(repeating: 0.0, count: inputs.count))

    var start = DispatchTime.now().uptimeNanoseconds
    let a = inputs.map(f)
    let t1 = Double(DispatchTime.now().uptimeNanoseconds - start) / 1e9

    start = DispatchTime.now().uptimeNanoseconds
    DispatchQueue.concurrentPerform(iterations: inputs.count) { i in
        let value = f(inputs[i])
        results.withLock { $0[i] = value }
    }
    let tp = Double(DispatchTime.now().uptimeNanoseconds - start) / 1e9
    let b = results.withLock { $0 }

    print("Sequential: \(t1) s");
    print("Parallel: \(tp) s");
    print("Speedup: \(t1 / tp)");
    print("Same results: \(a == b)");
}

demo()